### PR TITLE
chore(parser): delete redundancy reserved function token

### LINKED ISSUE #5076: Combine datetime interval add&subsruct  to one DATE_ADD
**Summary**

Syntax:
```
​​DATE_ADD('<unit>', <interval>, <date_or_time_expr>)
```

`<unit>`: `SECOND`, `MINUTE`, `HOUR`, `DAY`, `WEEK`, `MONTH`, `YEAR`
`<interval>`: Number, this can be negative for substract
`<date_or_time_expr>`:  The expression evaluate to a date, time, or timestamp

Example:
```
SELECT toDate(18875), DATE_ADD('YEAR', 2, toDate(18875));
+---------------+-----------------------------+
| 2021-09-05    | 2023-09-05                  |
+---------------+-----------------------------+

SELECT toDate(18875), DATE_ADD('YEAR', -2, toDate(18875));
+---------------+-----------------------------+
| 2021-09-05    | 2019-09-05                  |
+---------------+-----------------------------+
```

### PR DESCRIPTION
**Pull request by fkuner**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

It is not necessary to set 'YEAR' to reserved function keyword.

## Changelog

- Improvement

## Related Issues

Fixes #5076 #5419

### CODE REVIEW

**Review by andylokandy — APPROVED** [review]

File: common/ast/src/parser/token.rs [hunk 1]
```diff
@@ -792,14 +792,6 @@ impl TokenKind {
             // | TokenKind::WINDOW
             | TokenKind::WITH
             | TokenKind::DATEADD
```
Comment (andylokandy) [inline common/ast/src/parser/token.rs:794]: Maybe it'd be better if we rename it to `DATE_ADD`?

Comment (fkuner) [reply common/ast/src/parser/token.rs:794]: OK

**Comment by fkuner** [comment]
It seems work too if I remove the token`DATE_ADD` from reserved function. Is it good? @andylokandy

**Comment by andylokandy** [comment]
> It seems work too if I remove the tokenDATE_ADD from reserved function. Is it good? 

It's because after a refactor, the privilege of `DataAdd` became higher than `FunctionCall`. Anyway, I prefer to keep `DATE_ADD` reserved.

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**